dask_ml/impute.py

```diff
@@ -70,12 +70,18 @@ def _fit_frame(self, X):
         if self.strategy == "mean":
             avg = X.mean(axis=0).values
         elif self.strategy == "median":
-            avg = X.quantile().values
+            avg = [np.median(X[col].dropna()) for col in X.columns]
```

Comment:
I believe this will eagerly compute the values, thanks to `np.median`. Since that's done in a list comprehension, we'd end up executing the graph for `X` once per column. We want to delay computation till the end.

I also think this will end up pulling all the data for a column into a single ndarray, to do the median, which we also want to avoid.

Reply:
How about using `delayed` here? 
``` python
avg = [dask.delayed(np.median(X[col].dropna())) for col in X.columns]
```